frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1040,11 +1040,17 @@ func dispatchRun(d *dispatchState, c *instructions.RunCommand, proxy *llb.ProxyE
 }
 
 func dispatchWorkdir(d *dispatchState, c *instructions.WorkdirCommand, commit bool, opt *dispatchOpt) error {
-	d.state = d.state.Dir(c.Path)
-	wd := c.Path
-	if !path.IsAbs(c.Path) {
-		wd = path.Join("/", d.image.Config.WorkingDir, wd)
+	withoutDriveLetter, err := system.CheckSystemDriveAndRemoveDriveLetter(c.Path)
+	if err != nil {
+		return errors.Wrap(err, "removing drive letter")
+	}
+
+	d.state = d.state.Dir(withoutDriveLetter)
```

Comment:
This scans odd to me. I'm not sure why we have two separate ways for setting the path - the path in the state and the path in the config should be equivalent (this is also in the original).

I think we should restructure this so that we pass the result of `NormalizeWorkdir` directly into `d.state.Dir` (I think this is equivalent in behavior as long as `NormalizeWorkdir` always returns an absolute path?). Then we don't have two separate ways of computing the working directory.

If we're doing that, I think we can also push down the call to `CheckSystemDriveAndRemoveDriveLetter` into `NormalizeWorkdir`? Ideally, it would be nice to have as little windows-specific stuff in the dockerfile frontend code, since it then makes it harder for other frontends to also support windows.